### ISSUE TITLE
Difference between CHARMM build and AMBER build

### ISSUE THREAD

**Issue by AdriaPerezCulubret** [Acellera/htmd#97]
I'm working with the protein '2OV5', and when building the system for it, there's a protonated Aspartate (ASP246) that charmm build is failing to build properly, where amber is working fine.

I attached 3 images. The first one shows the crystal structure of the protein (the PDB file directly). It shows two Aspartates (ASP233 and 246) and it seems the one at right (ASP246) is making an hydrogen bond with the ASP at left (ASP233). The red dot alone is a water molecule.
![crystal_2ov5](https://cloud.githubusercontent.com/assets/15156429/17705239/fd22c718-63d7-11e6-966f-6c3b1fffbeee.png)

When building with Charmm, and after an equilibration process, the output looks like this:
![charmmbuild_2ov5](https://cloud.githubusercontent.com/assets/15156429/17705353/591e12ac-63d8-11e6-9a3a-79e63b2f50b4.png)

ASH246 gets a weird position, and the hydrogen added at the protonation gets an even more weird position. The interaction with ASP233 is lost.

Now, when building with Amber and equilibrating, the system looks more much alike with the crystal structure:
![amberbuild_2ov5](https://cloud.githubusercontent.com/assets/15156429/17705450/b5691732-63d8-11e6-92bd-64da3d9078eb.png)

Could this be an issue of charmm building? Or something related with different force fields used?

**Comment by j3mdamas** [comment]
Hi Adria,

I'll look into it.

Just to be sure, is this the same issue as Acellera/acemd_issues#10? If so, can we close that one (and thank you for ruling out as a build problem)?

**Comment by AdriaPerezCulubret** [comment]
Yes, it is the same issue, we can close it

El 16 ago. 2016 17:58, "João M. Damas" <notifications@github.com> escribió:

>

> >

> Reopened #97 <https://github.com/Acellera/htmd/issues/97>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
<https://github.com/Acellera/htmd/issues/97#event-757263111>, or mute the
thread

**Comment by j3mdamas** [comment]
thanks!

**Comment by j3mdamas** [comment]
adria is dumb

**Comment by j3mdamas** [comment]
I'll look into the forcefield

**Comment by j3mdamas** [comment]
This is probably a very serious issue. Confirmed on a completely independent build.

**Comment by tonigi** [comment]
Have a look if it is the same of http://www.ks.uiuc.edu/Research/namd/mailing_list/namd-l.2009-2010/2658.html

**Comment by stefdoerr** [comment]
Definitely the same. Thanks!

**Comment by j3mdamas** [comment]
yes, it is, but I'm researching.

the regenerate angles and dihedrals may generate duplicate parameters so I need to.

**Comment by j3mdamas** [comment]
this is fixed by multiscalelab/htmd@5cac3621f834a79012a80a17904c94bfea8e12ca

I still need to test this tomorrow:
- in the more general context of patching (there are patches that are made to fix angles and dihedrals, so these would be override by the fix)
- there was a warning on running the original charmm `autoregenerate angles dihedrals` with waters on the system, but this is psfgen, so I have to check if it applies to this as well

**Comment by stefdoerr** [comment]
So a single call to regenerate at the end of the script fixes the issue? That's nice

**Comment by j3mdamas** [comment]
Yes, but I need to check the things I said

**Comment by j3mdamas** [comment]
Furthermore, the build failed on the `charmm.py` test: https://travis-ci.org/multiscalelab/htmd/builds/155109024

I also have to check the differences there, probably caused by the regenerate.

**Comment by j3mdamas** [comment]
autopsf of vmd generates the correct output without the regenerate clause. I went to see the tcl script corresponding to the graphical interface, but I have no idea how is it able to do that without that clause. I ran the `build.vmd` on vmd instead of psfgen and it still gives the same wrong output, so it's not that...

I am running out of clues and people from the lab have used the regenerate clause in the past. For me, it's the solution to go.
